core/src/main/java/bio/terra/pearl/core/service/participant/WithdrawnEnrolleeService.java

```diff
@@ -1,106 +1,137 @@
-      //if a participant is withdrawing all of their relations are going to end too,
-      // otherwise the foreign key constraint will fail when deleting from enrollee table
-      // TODO in future, we may want to allow for a participant to
-      //  transfer their governed enrollees to another person before withdrawing
-      List<Enrollee> targetEnrolleesOnlyProxiedByEnrollee = enrolleerelationService.findExclusiveProxiedEnrollees(enrollee.getId());
-      enrolleerelationService.deleteAllByEnrolleeIdOrTargetId(enrollee.getId());
-      enrolleeService.delete(enrollee.getId(), CascadeProperty.EMPTY_SET);
-      //now withdraw all of the governed users that didn't have any other proxies
-        for (Enrollee targetEnrollee : targetEnrolleesOnlyProxiedByEnrollee) {
-            withdrawEnrollee(targetEnrollee);
-        }
+    private final EnrolleeService enrolleeService;
+    private final EnrolleeRelationService enrolleeRelationService;
+    private final ObjectMapper objectMapper;
+    private final PortalParticipantUserService portalParticipantUserService;
+    private final ParticipantUserService participantUserService;
+    private final EnrollmentService enrollmentService;
+    private final StudyEnvironmentService studyEnvironmentService;
+    private final StudyService studyService;
+
+    public WithdrawnEnrolleeService(WithdrawnEnrolleeDao dao, EnrolleeService enrolleeService, ObjectMapper objectMapper,
+                                    PortalParticipantUserService portalParticipantUserService, ParticipantUserService participantUserService,
+                                    EnrolleeRelationService enrolleeRelationService, EnrollmentService enrollmentService, StudyEnvironmentService studyEnvironmentService, StudyService studyService) {
+        super(dao);
+        this.enrolleeService = enrolleeService;
+        this.objectMapper = objectMapper;
+        this.portalParticipantUserService = portalParticipantUserService;
+        this.participantUserService = participantUserService;
+        this.enrolleeRelationService = enrolleeRelationService;
+        this.enrollmentService = enrollmentService;
+        this.studyEnvironmentService = studyEnvironmentService;
+        this.studyService = studyService;
+    }
+
+    public void deleteByStudyEnvironmentId(UUID studyEnvironmentId) {
+        dao.deleteByStudyEnvironmentId(studyEnvironmentId);
+    }
+
+    public int countByStudyEnvironmentId(UUID studyEnvironmentId) {
+        return dao.countByStudyEnvironmentId(studyEnvironmentId);
+    }
 
-      return withdrawnEnrollee;
-    } catch (JsonProcessingException e) {
-      throw new InternalServerException("Error serializing enrollee or user data", e);
+    public boolean isWithdrawn(String shortcode) {
+        return dao.isWithdrawn(shortcode);
     }
-  }
-
-  @Transactional
-  public List<WithdrawnEnrollee> withdrawFromPortal(PortalParticipantUser ppUser) throws JsonProcessingException {
```

Comment:
Were there any plans for this method (+ `withdrawFromJuniper` below)? They were unused, so I went ahead and removed them.